synapse/federation/federation_client.py

```diff
@@ -453,8 +459,9 @@ def send_request(destination):
 
             # Note: If not supplied, the room version may be either v1 or v2,
             # however either way the event format version will be v1.
-            room_version = ret.get("room_version", RoomVersions.V1.identifier)
-            event_format = room_version_to_event_format(room_version)
+            room_version_id = ret.get("room_version", RoomVersions.V1.identifier)
+            room_version = KNOWN_ROOM_VERSIONS.get(room_version_id)
```

Comment:
this could conceivably return `None`? in which case that should be documented in the docstring?

Or it should be changed to `KNOWN_ROOM_VERSIONS[room_version_id]`

Reply:
It now raises an `UnsupportedRoomVersionError`

Reply:
could probably do with putting that in the docstring too.